lib/validate_style.js

```diff
@@ -0,0 +1,47 @@
+'use strict';
+
+var validateConstants = require('./validate/validate_constants');
+var validate = require('./validate/validate');
+var latestStyleSpec = require('../reference/latest.min');
+var parseStyle = require('./parse_style');
```

Comment:
The previous file and export structure was carefully chosen and I'd prefer not to change it.

In particular, `validate_style.js` needs to _not_ `require('./parse_style')`, because that pulls in the entirety of the module and it's dependencies, which will bloat mapbox-gl-js.

See https://github.com/cutting-room-floor/mapbox-gl-style-lint/issues/48.